hadoop-ozone/client/src/test/java/org/apache/hadoop/ozone/client/MockOmTransport.java

```diff
@@ -116,13 +117,24 @@ public OMResponse submitRequest(OMRequest payload) throws IOException {
       return response(payload,
           r -> r.setServiceListResponse(
               serviceList(payload.getServiceListRequest())));
+    case AllocateBlock:
+      return response(payload, r -> r.setAllocateBlockResponse(
+          allocateBlock(payload.getAllocateBlockRequest())));
     default:
       throw new IllegalArgumentException(
           "Mock version of om call " + payload.getCmdType()
               + " is not yet implemented");
     }
   }
 
+  private OzoneManagerProtocolProtos.AllocateBlockResponse allocateBlock(
+      OzoneManagerProtocolProtos.AllocateBlockRequest allocateBlockRequest) {
+    return OzoneManagerProtocolProtos.AllocateBlockResponse.newBuilder()
+        .setKeyLocation(
+            blockAllocator.allocateBlock(allocateBlockRequest.getKeyArgs())
+                .iterator().next()).build();
```

Comment:
Let me add loop here for all KeyArgs available

Reply:
Looking at SinglePipelineBlockAllocator, I am not sure it can return multiple results, or that you can request multiple blocks in one go from the AllocateBlockRequest. However it does return an Iterable<>, so you have to call next on it as you have done.

The response AllocateBlockResponse object just contains a single "KeyLocation", which contains a single "BlockID" and "Pipeline". Therefore I *think* your code is correct and it does not need a loop here.

Reply:
oh yeah. Looks like your are right. API name says keyLocation, but not keyLocations.

Reply:
It's possible to return with multiple key locations if the keyArgs contains a size which is huge enough. Today it's not supported by the block allocated, but looks to be safer to add all the results to the `setKeyLocation`...

Reply:
Oh, thanks for the offline discussion with you, I got it. Pure `allocateBlock` supports one `keyLocation`. Open key supports multiple location. Strange,  but this is what we have. Thanks to help me learninig it.